signalhandler_unix.go

```diff
@@ -0,0 +1,100 @@
+// +build !windows
+
+package endless
+
+import (
+	"os"
+	"os/signal"
+	"syscall"
+)
+
+// The default Handler for none Windows OS's is the a SignalHandler
+func init() {
+	DefaultHandler = NewSignalHandler()
+}
+
+// SignalHook represents a signal processing hook.
+// If false is returned no further processing of the signal is performed.
+type SignalHook func(sig os.Signal, srv *Server) (cont bool)
+
+// SignalHandler listens for signals and takes action on the Server.
+//
+// By default:
+// SIGHUP: calls Restart()
+// SIGUSR2: calls Terminate(0), Shutdown() must have been called first.
+// SIGINT & SIGTERM: calls Shutdown()
+//
+// Pre and post signal handles can also be registered for custom actions.
+type SignalHandler struct {
+	preHooks  map[os.Signal][]SignalHook
+	postHooks map[os.Signal][]SignalHook
+}
+
+// NewSignalHandler create a new SignalHandler for the s
+func NewSignalHandler() *SignalHandler {
+	return &SignalHandler{
+		preHooks:  make(map[os.Signal][]SignalHook),
+		postHooks: make(map[os.Signal][]SignalHook),
+	}
+}
+
+// Handle listens for os.Signal's and calls any registered function hooks.
```

Comment:
actually nevermind, since we're talking about a programming language symbol, rather than an actual word, the same rules arguably don't apply, so an apostrophe is fine here.